### PR TITLE
Add info for Julia version used for testing

### PR DESCRIPTION
**Pull request by navidcy**
PR stems from https://github.com/CliMA/Oceananigans.jl/commit/d8273e7a07e8f53f9772f87f67f6eca8d17cb98c#commitcomment-150654994

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -24,8 +24,11 @@ julia> using Pkg
 julia> Pkg.add("Oceananigans")
 ```
 
-!!! compat "Julia 1.9 or later is required; Julia 1.10 is suggested"
-    Oceananigans requires Julia 1.9 or later. We strongly urge you to use Julia 1.10.
+!!! compat "Julia 1.9 or later is required"
+    Oceananigans requires Julia 1.9 or later.
+    
+!!! info "Continuous integration testing"
```
Comment (glwagner) [inline docs/src/index.md:30]:
```suggestion
!!! info "Tested Julia versions"
```

File: docs/src/index.md [hunk 2]
```diff
@@ -24,8 +24,11 @@ julia> using Pkg
...
+    Oceananigans is tested on Julia 1.10.
```
Comment (glwagner) [inline docs/src/index.md:31]:
```suggestion
    Oceananigans is currently tested on Julia 1.10.
```

Comment (navidcy) [reply docs/src/index.md:31]: What does "currently" add here? The docs are associated with package versions; later on for version 0.423.2 we may change this to Julia 1.12. Right?

Comment (glwagner) [reply docs/src/index.md:31]: My intent is that if someone notices it is wrong it gives them a clue why and they can submit a PR to fix it

**Review by simone-silvestri — APPROVED** [review]

**Pull request merged**